apps/journeys-admin/src/components/Editor/Slider/Settings/CanvasDetails/Properties/blocks/Card/BackgroundColor/BackgroundColor.tsx

```diff
@@ -132,6 +137,28 @@ export function BackgroundColor(): ReactElement {
     }
   }
 
+  function isValidHex(color: string): boolean {
+    const hexColorRegex = /^#[0-9A-Fa-f]{6}$/
+    return hexColorRegex.test(color)
+  }
+
+  async function handleFieldChange(color: string): Promise<void> {
+    if (isValidHex(color)) {
+      await handleColorChange(color)
+    }
+    if (color.length < 8) {
+      setColor(color)
+    }
+  }
+
+  async function handleBlur(color: string): Promise<void> {
+    if (isValidHex(color)) {
+      await handleColorChange(color)
+    } else {
+      setColor(selectedColor)
+    }
+  }
```

Comment:
```suggestion
  const validationSchema = object({
    color: string()
      .required(t('Required'))
      .test('valid-hex-color', t('Invalid HEX color code'), isValidHex)
  })
```

Drop these functions and use `Formik` to handle validation, see `LinkAction.tsx`